libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/impl/EmailNotificationService.java

```diff
@@ -134,7 +148,10 @@ String getBody(String type, Map<String, String> details) {
                 body = getPrincipalExpiryBody(details);
                 break;
         }
-        body = body + getFooter();
+        if (addFooter) {
+            body = body + getFooter();
+        }
+        body = body.replace("<style></style>", "<style>" + readContentFromFile(EMAIL_TEMPLATE_CSS) + "</style>");
```

Comment:
here and also on line 193, we seem to be calling readContentsFromFile every time we generate a message. Can't we read those contents only once during class constructor and then keep reusing them?